src/builtin_unset.c

```diff
@@ -0,0 +1,40 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   builtin_unset.c                                    :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: gwolf <gwolf@student.42vienna.com>         +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/07/25 13:07:27 by gwolf             #+#    #+#             */
+/*   Updated: 2023/07/26 22:21:47 by gwolf            ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+/**
+ * @file builtin_unset.c
+ * @brief Implementation of builtin unset.
+ */
+#include "mod_builtin.h"
+
+/**
+ * @brief Delete variables out of environment.
+ *
+ * For every arg try ft_hashtable_delete().
+ * Returns always SUCCESS, even if var was not found.
+ *
+ * @param argv NULL terminated args.
+ * @param env_tab Environment.
+ * @return t_err SUCCESS.
+ */
+t_err	ft_unset(char **argv, t_hashtable *env_tab)
+{
+	size_t		i;
+
+	i = 1;
+	while (argv[i])
+	{
+		ft_hashtable_delete(env_tab, argv[i], ft_strlen(argv[i]));
+		i++;
+	}
```

Comment:
No.

Reply:
'tis not a valid answer

Reply:
For me this change would negatively impact readability for no real upside (at least I don't see one)

- The loop is already doing two things (delete and calc strlen), it would increase complexity
- At first glance - at least for me - it's confusing because the line is handling the same arg, but at same time at the last instance the index gets incremented. You have to use extra brain power to realize it's still the same arg, and incremented for the next run of the loop. 
- The function has plenty of lines left. I think the Norm doesn't want to enforce 'clever' written, hard(er) to understand code with its line limit. It wants to enforce the "do one thing, but do it good" paradigm for functions. Using clever code shortations should be used in rare cases, where the line limit is reached but no other measure can safely be taken. Ask yourself: is the one liner version -exactly- as easy to understand as the five liner? If not: Why shorten it?
- Performance is no issue: the compiler optimizes the code anyway. Even if not: the microseconds in gain are far outweighed by the seconds lost parsing the code (by the human), if any errors are to be investigated.

One could argue that a consistent use of the incrementer in these loops makes the eye more accustomed to it, and therefore increases human parsing speed. But that's kinda like saying we have to write the code in light blue color on dark blue background because with time our comprehension would increase. Why even do it in the first place?

Reply:
I think it is not that hard to read as you just follow the execution of the line from left to right and at the end realise that the counter gets incremented after being used in ft_strlen(). As you already pointed out it is a matter of getting accustomed to that kind of code "style". 

Personally I am an advocate of keeping the code as short as possible, especially in cases of if conditions or while loops where the brackets can be saved by having just one line of code within. That facilitates a faster browsing of the code for later maintenance and simply improves coding speed as there is less to type. 

I guess it's a question of keeping the balance between short and sharp code and readability. IMO the intention of the code would rest clear enough with that change, but I leave it to your discretion.

Reply:
Then I will leave it :D

I think if there would be a bug to trace I would at least second guess myself if I'm really incrementing here correctly. I don't think I can reach any level of accustomisation where the two would be equally parsed by me.